examples/training/object_detection/pascal_voc/retina_net.py

```diff
@@ -157,18 +157,13 @@ def unpackage_inputs(data):
 # TODO(lukewood): the boxes loses shape from KPL, so need to pad to a known shape.
 # TODO(tanzhenyu): consider remove padding while reduce function tracing.
 def pad_fn(image, boxes):
-    boxes = boxes.to_tensor(default_value=-1.0, shape=[GLOBAL_BATCH_SIZE, 32, 5])
-    gt_boxes = boxes[..., :4]
-    gt_classes = boxes[..., 4]
-    return image, {
-        "boxes": gt_boxes,
-        "classes": gt_classes,
-    }
+    return image, bounding_box.to_dense(boxes)
```

Comment:
My suggestion is don't offer a last-minute API before 0.4 release, instead just keep the padding function as-is, given this is a training example script

Reply:
the padding function doesn't work with the dictionary format.   We need this API no matter what internally - this is a very common conversion.  It replaces filter_sentinels and pad_with_sentinels.